app/views/api/v2/models/main.json.rabl

```diff
@@ -3,3 +3,10 @@ object @model
 extends "api/v2/models/base"
 
 attributes :info, :created_at, :updated_at, :vendor_class, :hardware_model
+
+if params.has_key?(:show_permissions)
+  node(:can_edit) { |model| model.authorized?(:edit_model)}
+  node(:can_delete) { |model| model.authorized?(:delete_model)}
+end
+
+node(:hosts_count) { |model| model.hosts.length }
```

Comment:
this is wrong, it will load the host objects or do N+1, I understand why you did the preloading now, but this can be fetched with one SQL query (as I commented earlier).